packages/fetchai/skills/tac_control_contract/behaviours.py

```diff
@@ -171,36 +183,47 @@ def _request_mint_items_transaction(self, game: Game) -> None:
 
         :return: None
         """
-        self.context.logger.info("requesting mint_items transactions.")
-        for agent_state in game.initial_agent_states.values():
-            parameters = cast(Parameters, self.context.parameters)
-            token_ids = []  # type: List[int]
-            mint_quantities = []  # type: List[int]
-            for good_id, quantity in agent_state.quantities_by_good_id.items():
-                token_ids.append(int(good_id))
-                mint_quantities.append(quantity)
-            for currency_id, amount in agent_state.amount_by_currency_id.items():
-                token_ids.append(int(currency_id))
-                mint_quantities.append(amount)
-            contract_api_dialogues = cast(
-                ContractApiDialogues, self.context.contract_api_dialogues
-            )
-            contract_api_msg, contract_api_dialogue = contract_api_dialogues.create(
-                counterparty=LEDGER_API_ADDRESS,
-                performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,
-                ledger_id=parameters.ledger_id,
-                contract_id="fetchai/erc1155:0.10.0",
-                contract_address=parameters.contract_address,
-                callable=ContractApiDialogue.Callable.GET_MINT_BATCH_TRANSACTION.value,
-                kwargs=ContractApiMessage.Kwargs(
-                    {
-                        "deployer_address": self.context.agent_address,
-                        "recipient_address": self.context.agent_address,
-                        "token_ids": token_ids,
-                        "mint_quantities": mint_quantities,
-                    }
-                ),
-            )
-            contract_api_dialogue = cast(ContractApiDialogue, contract_api_dialogue)
-            contract_api_dialogue.terms = parameters.get_mint_token_terms()
-            self.context.outbox.put_message(message=contract_api_msg)
+        if not game.is_allowed_to_mint:
+            return
+        game.is_allowed_to_mint = False
+        agent_state = game.get_next_agent_state_for_minting()
+        if agent_state is None:
+            return
+        name = game.registration.agent_addr_to_name[agent_state.agent_address]
+        self.context.logger.info(
+            f"requesting mint_items transactions for agent={name}."
+        )
+        parameters = cast(Parameters, self.context.parameters)
+        token_ids = []  # type: List[int]
+        mint_quantities = []  # type: List[int]
```

Comment:
We can move type annotation in front of the variables, e.g. we can write `token_ids: List[int] = []`